examples/bigquery_view_parser.py

```diff
@@ -67,106 +67,112 @@ def _get_parser(cls):
+            VARIANCE, WHEN, WHERE, WINDOW, WITH, WITHIN
+                 """.replace(",", "").split())
+        # fmt: on
 
-        keyword_nonfunctions = MatchFirst(
-            (UNION, ALL, INTERSECT, EXCEPT, COLLATE, ASC, DESC, ON, USING,
-             NATURAL, INNER, CROSS, LEFT, RIGHT, OUTER, FULL, JOIN, AS, INDEXED,
-             NOT, SELECT, DISTINCT, FROM, WHERE, GROUP, BY, HAVING, ORDER, BY,
-             LIMIT, OFFSET, CAST, ISNULL, NOTNULL, NULL, IS, BETWEEN, ELSE, END,
-             CASE, WHEN, THEN, EXISTS, COLLATE, IN, LIKE, GLOB, REGEXP, MATCH,
-             STRUCT, WINDOW,
-             )
-        )
+        keyword = MatchFirst((
+            ALL, AND, ANY, ARRAY, AS, ASC, ASSERT_ROWS_MODIFIED, AT, BETWEEN, BY, CASE, CAST,
+            COLLATE, CONTAINS, CREATE, CROSS, CUBE, CURRENT, DEFAULT, DEFINE, DESC, DISTINCT,
+            ELSE, END, ENUM, ESCAPE, EXCEPT, EXCLUDE, EXISTS, EXTERNAL_QUERY, EXTRACT, FALSE,
+            FETCH, FOLLOWING, FOR, FROM, FULL, GROUP, GROUPING, GROUPS, HASH, HAVING, IF,
+            IGNORE, IN, INNER, INTERSECT, INTERVAL, INTO, IS, JOIN, LATERAL, LEFT, LIKE, LIMIT,
+            LOOKUP, MERGE, NATURAL, NEW, NO, NOT, NULL, NULLS, OF, ON, OR, ORDER, OUTER, OVER,
+            PARTITION, PIVOT, PRECEDING, PROTO, RANGE, RECURSIVE, RESPECT, RIGHT, ROLLUP, ROWS,
+            SELECT, SET, SOME, STRUCT, TABLESAMPLE, THEN, TO, TREAT, TRUE, UNBOUNDED, UNION,
+            UNNEST, USING, WHEN, WHERE, WINDOW, WITH, WITHIN))
 
-        keyword = keyword_nonfunctions | MatchFirst(
-            (ESCAPE, CURRENT_TIME, CURRENT_DATE, CURRENT_TIMESTAMP, DATE_ADD,
-             DATE_SUB, ADDDATE, SUBDATE, INTERVAL, STRING_AGG, REGEXP_EXTRACT,
-             SPLIT, ORDINAL, UNNEST, SAFE_CAST, PARTITION, TIMESTAMP_ADD,
-             TIMESTAMP_SUB, ARRAY, GENERATE_ARRAY, GENERATE_DATE_ARRAY,
-             GENERATE_TIMESTAMP_ARRAY,
-             )
-        )
-        # fmt: on
+        # those are keywords that are fiunction names
+        keyword_funcs = MatchFirst((
+            TREAT, WITHIN, TABLESAMPLE, SOME, SET, RIGHT, PROTO, MERGE,
+            LOOKUP, LEFT, LATERAL, FETCH, CURRENT
+        ))
 
-        identifier_word = Word(alphas + "_@#", alphanums + "@$#_")
+        identifier_word = Word(alphas + '_@#', alphanums + '@$#_')
         identifier = ~keyword + identifier_word.copy()
         collation_name = identifier.copy()
-        # NOTE: Column names can be keywords.  Doc says they cannot, but in practice it seems to work.
-        column_name = identifier_word.copy()
-        qualified_column_name = Combine(
-            column_name + ("." + column_name)[..., 6], adjacent=False
-        )
+        # NOTE: Column names can't be keywords unless they are quoted
+        column_name = identifier.copy() | Suppress('`') + identifier_word + Suppress('`')
+        # first part of multi part column name can't be keyword, other parts can
+        qualified_column_name = Combine(column_name + (
```

Comment:
Instead of adding `ZeroOrMore(' ')`, can you do just do `Combine` with `adjacent=False`? (What if there is a line break instead of a space, for instance?)